ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/ChainDB/StateMachine.hs

```diff
@@ -1767,56 +1768,75 @@ genBlk chunkInfo Model{..} =
         )
       ]
 
+genSecurityParam :: Gen SecurityParam
+genSecurityParam =
+  SecurityParam
+    . unsafeNonZero
+    . fromIntegral
+    . (+ 2) -- shift to the right to avoid degenerate cases
+    <$> geometric 0.5 -- range in [0, +inf); mean = 1/p = 2
```

Comment:
Maybe here it'd be a good point to explain why we chose a geometric distribution for generating `k`.

Reply:
Added an adapted version of the analysis we included in the commit message of https://github.com/IntersectMBO/ouroboros-consensus/pull/1772/commits/842c8a9660a776644f3078e99262dd59cd79cba3 in https://github.com/IntersectMBO/ouroboros-consensus/pull/1772/commits/d521432cf90f1035a59f64e241cdb30189206642